apps/src/lab2/views/dialogs/GenericPrompt.tsx

```diff
@@ -67,18 +74,60 @@ const GenericPrompt: React.FunctionComponent<GenericPromptProps> = ({
     undefined
   );
 
+  const debouncedErrorHandler = useMemo(() => {
+    return debounce((newInput: string) => {
+      setErrorMessage(validateInput(newInput));
+    }, DEBOUNCE_TIME_OUT);
+  }, [setErrorMessage, validateInput]);
+
   const handleInputChange = useCallback(
     (newInput: string) => {
       setPromiseArgs(newInput);
-      setErrorMessage(validateInput(newInput));
+      if (newInput.length && !errorMessage?.length) {
+        debouncedErrorHandler(newInput);
+      } else {
+        setErrorMessage(validateInput(newInput));
+      }
```

Comment:
if the user has typed something in _and_ we do not currently have an error, then use the debounced handler with the delay. That'll prevent the error from popping up immediately and giving a chance to type.

Otherwise, if there is no input _or_ the user already has an error, then we want to validate immediately (in an attempt to clear the error), so use the non-debounced version.

Reply:
It might be worth adding a comment with this explanation here